server/planning/planning_article_export.py

```diff
@@ -0,0 +1,131 @@
+
+import superdesk
+
+from flask import render_template_string, current_app
+
+from apps.auth import get_user_id
+from apps.templates.content_templates import get_item_from_template
+from apps.archive.common import insert_into_versions
+from copy import deepcopy
+import json
+
+
+class PlanningArticleExportResource(superdesk.Resource):
+    schema = {
+        'items': {
+            'type': 'list',
+            'required': True,
+        },
+        'desk': superdesk.Resource.rel('desks', required=True),
+        'template': {'type': 'string'},
+        'type': {
+            'type': 'string',
+            'default': 'planning',
+        }
+    }
+
+    item_methods = []
+    resource_methods = ['POST']
+    privileges = {'POST': 'planning'}
+
+
+def get_item(_id, resource):
+    item = superdesk.get_resource_service(resource).find_one(None, _id=_id) or {}
+    if item.get('event_item'):
+        item['event'] = superdesk.get_resource_service('events').find_one(None, _id=item['event_item'])
+        if item['event'] and item['event'].get('event_contact_info'):
+            item['contacts'] = []
+            contacts = superdesk.get_resource_service('contacts').find(where={
+                '_id': {'$in': (item['event']['event_contact_info'])}
+            })
+            for contact in contacts:
+                if contact.get('public'):
+                    item['contacts'].append(contact)
+    elif resource == 'events':
+        item['plannings'] = superdesk.get_resource_service('events').get_plannings_for_event(item)
```

Comment:
At the moment `contacts` are only exported when exporting a Planning item associated with an Event.

It would be better if `contacts` were also exported when exporting an Event.